unitylibs/core/workflow/workflow-upload/action-binder.js

```diff
@@ -0,0 +1,359 @@
+    this.canvasArea.forEach((element) => {
+      element.style.pointerEvents = 'none';
+      const errorToast = element.querySelector('.alert-holder');
+      const closeBtn = errorToast?.querySelector('.alert-close');
+      if (closeBtn) closeBtn.style.pointerEvents = 'auto';
+      if (!errorToast) return;
+      const alertText = errorToast.querySelector('.alert-text p');
+      if (!alertText) return;
+      alertText.innerText = msg;
+      errorToast.classList.add('show');
+    });
+  }
+}
+
+export default class ActionBinder {
+  constructor(unityEl, workflowCfg, wfblock, canvasArea, actionMap = {}) {
+    this.unityEl = unityEl;
+    this.workflowCfg = workflowCfg;
+    this.block = wfblock;
+    this.actionMap = actionMap;
+    this.canvasArea = canvasArea;
+    this.errorToastEl = null;
+    this.psApiConfig = this.getPsApiConfig();
+    this.serviceHandler = null;
+    this.splashScreenEl = null;
+    this.transitionScreen = null;
+    this.LOADER_LIMIT = 95;
+    this.limits = workflowCfg.targetCfg.limits;
+    this.promiseStack = [];
+    this.initActionListeners = this.initActionListeners.bind(this);
+  }
+
+  getPsApiConfig() {
+    unityConfig.psEndPoint = {
+      assetUpload: `${unityConfig.apiEndPoint}/asset`,
+      acmpCheck: `${unityConfig.apiEndPoint}/asset/finalize`,
+    };
+    return unityConfig;
+  }
+
+  async handlePreloads() {
+    const parr = [];
+    if (this.workflowCfg.targetCfg.showSplashScreen) {
+      parr.push(
+        `${getUnityLibs()}/core/styles/splash-screen.css`,
+      );
+    }
+    await priorityLoad(parr);
+  }
+
+  async cancelUploadOperation() {
+    try {
+      document.querySelector('a.con-button[href*="#_cancel"]').setAttribute('daa-ll', 'cancel');
+      const { default: TransitionScreen } = await import(`${getUnityLibs()}/scripts/transition-screen.js`);
+      this.transitionScreen = new TransitionScreen(this.transitionScreen.splashScreenEl, this.initActionListeners, this.LOADER_LIMIT, this.workflowCfg);
+      await this.transitionScreen.showSplashScreen();
+      const e = new Error('Operation termination requested.');
+      const cancelPromise = Promise.reject(e);
+      this.promiseStack.unshift(cancelPromise);
+    } catch (error) {
+      await this.transitionScreen?.showSplashScreen();
+      throw error;
+    }
+  }
+
+  extractFiles(e) {
+    const files = [];
+    if (e.dataTransfer?.items) {
+      [...e.dataTransfer.items].forEach((item) => {
+        if (item.kind === 'file') files.push(item.getAsFile());
+      });
+    } else if (e.target?.files) {
+      [...e.target.files].forEach((file) => files.push(file));
+    }
+    return files;
+  }
+
+  async uploadImgToUnity(storageUrl, id, blobData, fileType) {
+    const uploadOptions = {
+      method: 'PUT',
+      headers: { 'Content-Type': fileType },
+      body: blobData,
+    };
+    const response = await fetch(storageUrl, uploadOptions);
+    return response.status === 200 ? id : '';
```

Comment:
I see that we are using `console.error` elsewhere. Do we want to log the non-200 response + message for now?